p2p_wallet/Scenes/Main/History/History.Repository.swift

```diff
@@ -0,0 +1,79 @@
+//
+// Created by Giang Long Tran on 15.04.2022.
+//
+
+import Foundation
+
+/// The repository that works with transactions.
+protocol HistoryTransactionRepository {
+    /// Fetches a list of signatures, that belongs to the `address`.
+    ///
+    /// - Parameters:
+    ///   - address: the account address
+    ///   - limit: the number of transactions that will be fetched.
+    ///   - before: the transaction signature, that indicates the offset of fetching.
+    /// - Returns: the list of `SignatureInfo`
+    func getSignatures(address: String, limit: Int, before: String?) async throws -> [SolanaSDK.SignatureInfo]
+    
+    /// Fetch all data of the transaction
+    ///
+    /// - Parameter signature: The transaction signature
+    /// - Returns: `TransactionInfo`, that can be parsed later.
+    func getTransaction(signature: String) async throws -> SolanaSDK.TransactionInfo
+}
+
+extension History {
+    typealias TransactionRepository = HistoryTransactionRepository
+
+    class SolanaTransactionRepository: HistoryTransactionRepository {
+        private let solanaSDK: SolanaSDK
+
+        init(solanaSDK: SolanaSDK) { self.solanaSDK = solanaSDK }
+
+        func getSignatures(address: String, limit: Int, before: String?) async throws -> [SolanaSDK.SignatureInfo] {
+            try await solanaSDK
+                .getSignaturesForAddress(address: address, configs: .init(limit: limit, before: before))
+                .value
+        }
+
+        func getTransaction(signature: String) async throws -> SolanaSDK.TransactionInfo {
+            try await solanaSDK.getTransaction(transactionSignature: signature).value
+        }
+    }
+
+    class CachingTransactionRepository: TransactionRepository, Cachable {
```

Comment:
`CachingTransactionRepository` is not so needed, we can use some kind of PersistentStore for `SolanaTransactionRepository`